### DISCUSSION TITLE
Improve the Schedule view

### DISCUSSION THREAD

**Discussion by rogercreagh** [AzuraCast/AzuraCast#7591] [Ideas]
Restarting this as it got derailed by an inappropriate comment. I am very grateful for all the work the developer(s) do and know from personal experience it is simply not possible to please everyone with this sort of project.

Having said which I'd like to suggest a couple of improvements to the schedule view (and to the API to get schedule data)

1.Firstly. Rather than (or in addition to) a static view of the whole current week it would seem more useful to have a rolling 7 day view as the past info is of less interest to the listener and by Saturday you probably want to see what is scheduled on the following Monday, not the previous one.

2.Secondly the full 24 hr frame is very long - how about having a start and end time to display for the days. That way the station can present separate morning, afternoon a evening and night schedules (with an overlap if desired) which would be far more readable.

3.Thirdly  the full frame format is very cumbersome - as an alternative how about a view that simply listed the next N scheduled items.  (N being definable either by the end user or by the station). This would be much more useful for mobile devices where the text in the frame gets broken over lines and also for those stations that don't schedule a playlist every moment of the day but have gaps where it is just the general rotation playing. (eg all night)

4. Fourthly - these need to be made available as additional widget options so they can easily be embedded in a website.

5. Fifthly on the API - Yes I know you can probably do much of this using the API but not everyone is going to be able to hack that. And while we are on the subject of the API how about adding a parameter there for end datetime rather than the number of items to return - if you just want to get a week of schedule its a pain having to guess that there are no more than N items and then filter out the extra ones if necessary
Also it would be nice to be able to fetch just the schedule for one playlist (or maybe a list of playlists)

So 5 related suggestions to discuss and prioritise for eventual implementation. Thanks.
RogerCO

**Comment by eternityready2** [comment]
@rogercreagh i updated azura cast to have exended schedule options @ https://github.com/eternityready2/AzuraCast
